plasmapy/formulary/lengths.py

```diff
@@ -270,7 +289,22 @@ def gyroradius(
 
     omega_ci = frequencies.gyrofrequency(B, particle)
 
-    return np.abs(Vperp) / omega_ci
+    if np.isscalar(lorentzfactor):
+        # if it is not finite aka not defined then calc from vperp
+        if not isfinite_lorentzfactor:
+            lorentzfactor = np.copy(lorentzfactor)
+            lorentzfactor = RelativisticBody(particle, Vperp).lorentz_factor
+        # else it is defined already so you are all good and don't need to do anything
+    elif np.all(
+        isfinite_lorentzfactor
+    ):  # if it is not a scalar you want to calculate Lorentz factor based on Vperp for all undefined entries of the array
+        rbody = RelativisticBody(particle, Vperp)
+        lorentzfactor = np.copy(lorentzfactor)
+        lorentzfactor[not isfinite_lorentzfactor] = rbody.lorentz_factor()[
+            not isfinite_lorentzfactor
+        ]
```

Comment:
Hey, nice job providing an example of what I wrote before! 😜  Pretty sure you're actually going to need to do it like this, because `not` will try to negate the whole array object while `~` can act elementwise:
```suggestion
        lorentzfactor = np.copy(lorentzfactor)
        lorentzfactor[~isfinite_lorentzfactor] = rbody.lorentz_factor()[
            ~isfinite_lorentzfactor
        ]
```

Does RelativisticBody even work with non-scalar array values?